src/ng/directive/ngOptions.js

```diff
@@ -325,13 +325,25 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         // The option values were already computed in the `getWatchables` fn,
         // which must have been called to trigger `getOptions`
         var optionValues = valuesFn(scope) || [];
+        var optionValuesKeys;
 
-        var keys = Object.keys(optionValues);
-        keys.forEach(function getOption(key) {
 
-          // Ignore "angular" properties that start with $ or $$
-          if (key.charAt(0) === '$') return;
+        if (!keyName && isArrayLike(optionValues)) {
+          optionValuesKeys = optionValues;
+        } else {
+          // if object, extract keys, in enumeration order, unsorted
+          optionValuesKeys = [];
+          for (var itemKey in optionValues) {
```

Comment:
Yes I thought of all those options and I do agree. The reason I went with this is that it is much closer (almost line for line) with the `ngRepeat` implementation and I thought it would be best to keep the two as similar as possible.

I believe that inside `ngRepeat` the argument is to try to minimize function calls to keep things fast. I have no idea if that is a valid performance issue in this case. I am happy to take your advice on that.